docs/source/code.rst

```diff
@@ -0,0 +1,61 @@
+Code Guidelines
+-------------------------------------------
+
+Developers of eMach should read this document to understand expectations of code contributed to the eMach codebase. This document outlines the 
+code requirements for ``analyzers`` and ``machine_designs``. Understanding of proper Python coding practices are necessary to contribute to each 
+section of ``eMach``.
+
+Analyzer Code
+++++++++++++++++++++++++++++++++++++++++++++
+
+The code required for each ``analyzer`` within the ``eMach`` repository must be structured to analyze any set of user-defined inputs and produce 
+user-defined outputs. The code structure of each ``analyzer`` should contain each of the following classes:
+
+1. DesignProblem Class
+2. DesignAnalyzer Class
+
+DesignProbelm Class
+*******************************************
+
+The ``DesignProblem`` class takes all of the user-defined inputs as its arguments. The output of the ``DesignProblem`` class is a single variable with 
+multiple attributes. The code for each input variable must clearly defined the input variable and add it as an attribute of the `self` variable within 
+each ``DesignProblem`` class. The output of the ``DesignProblem`` class is the single `self` variable with as many attributes as desired. Commented out
+within the class should be explanations of each input variable. This should include the variable name, a brief description of the variable, and the 
+units of that variable. The output of the ``DesignProblem`` class is then used by the ``DesignAnalyzer` class.
+
+DesignAnalyzer Class
+*******************************************
+
+The ``DesignAnalyzer`` class is where all of the data processing required given the problem definition occurs. This class takes the singular output 
+variable from the ``DesignProbelm`` class defined as `problem: XXXProbelm`. The `XXX` term should be the title of the analyzer. Commented out within 
+the ``DesignAnalyzer`` class should be each of it's input arguments and output returns. The commented code should contain the variable named, brief 
+descriptions of the variables, and variable units split between `arguments` and `returns` sections.
+
```

Comment:
I think the two most important things to capture here for analyzers:
1. What is expected for the initializer
2. What is expected of the analyzer function (it's argument and what it returns). 

Somehow this isn't fully captured.